### PR TITLE
AMDGPU/NewPM: Port SIFixSGPRCopies to new pass manager

### PR DESCRIPTION
**Pull request by arsenm**
This allows moving some tests relying on -stop-after=amdgpu-isel
to move to checking -stop-after=finalize-isel instead, which
will more reliably pass the verifier.

### CODE REVIEW

**Comment by arsenm** [comment]
* **#102614** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/102614?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @arsenm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: llvm/lib/Target/AMDGPU/SIFixSGPRCopies.cpp [hunk 1]
```diff
@@ -158,6 +157,20 @@ class SIFixSGPRCopies : public MachineFunctionPass {
   bool tryMoveVGPRConstToSGPR(MachineOperand &MO, Register NewDst,
                               MachineBasicBlock *BlockToInsertTo,
                               MachineBasicBlock::iterator PointToInsertTo);
+};
+
+class SIFixSGPRCopiesLegacy : public MachineFunctionPass {
```
Comment (jayfoad) [inline llvm/lib/Target/AMDGPU/SIFixSGPRCopies.cpp:162]: I'd have called it SIFixSGPRCopiesLegacy**Pass** but I see there are precedents for this weird naming.

**Review by jayfoad — APPROVED** [review]
LGTM

File: llvm/lib/Target/AMDGPU/SIFixSGPRCopies.cpp (hunk 1 above)
Comment (arsenm) [reply llvm/lib/Target/AMDGPU/SIFixSGPRCopies.cpp:162]: If you do that, you end up with the "PassPass" suffix for the initialize* functions

**Pull request merged**